### ISSUE TITLE
Mapping

### ISSUE THREAD

**Issue by mcrascal** [evidence-dev/evidence#565]
Users have requested a variety of mapping solutions. This is a very large topic, so I propose we start relatively small here. 

Maps lend themselves well to a layer based interface using slots. Eventually, I think we should drive toward something like this: 

```

<Map data = {query} projection='mercator' zoom='auto'>
   <BaseMap .../> 
   <Markers ... />
   <HeatMap .../> 
</Map > 

```

But the most pressing current customer need is for a US State Choropleth. 

To start, I think we should implement this as a standalone component leveraging our existing ECharts infrastructure. For example:

```
<USStates .../> 
```

**Comment by hughess** [comment]
I took a quick look to see if I could get something spun up for the US state map using a similar pattern to the rest of our echarts library usage. 

It was simpler to start from scratch since there are some different echarts functions we need to run when initiating the chart area vs. our typical chart use cases, but there's likely a way to incorporate them all into the same set of functions.

Here's what I ended up with:
```html
<USMap data={map_data}/>
```

![Image](https://user-images.githubusercontent.com/12602440/214184054-3713aea0-1def-425a-826b-da8e7744b691.png)


![Image](https://user-images.githubusercontent.com/12602440/214184223-922d0128-078f-4487-91e7-99fb89083a72.gif)

## Details
- Requires two columns explicitly named `name` and `value`
- Requires the full state name. Can't use a two letter abbreviation, although I think it would be straightforward to enable that
- This relies on a JSON file with the coordinates of the state boundaries
- Not sure how to remove the visual map legend on the left hand side
- We would need to figure out how to incorporate the logic we already have set up for chart titles/subtitles, image download, and data download (maybe this is a good enough reason to get the map flowing through our existing ECharts component)
- I haven't looked into the theming options available for maps, but it would be nice to be able to customize the border colours and scale colours to make it look more Evidence-y

**Comment by hughess** [comment]
It's also relatively straightforward to add a world map or regional maps (however we'd like to define them), but it involves having a fairly large JSON file sitting in the project for each map we want to support.

Also played around with a custom colour scale and border style for this one.

<img width="585" alt="CleanShot 2023-01-24 at 09 56 34@2x" src="https://user-images.githubusercontent.com/12602440/214328326-e8c1f73b-67a3-41a9-a25c-a986cc247e21.png">

**Comment by hughess** [comment]
Making some progress here - playing around with some colour palettes for the first version:
<img width="581" alt="CleanShot 2023-01-25 at 14 01 44@2x" src="https://user-images.githubusercontent.com/12602440/214658434-2a09bdbe-d117-4bb1-84f7-13525dcb5d2e.png">
<img width="581" alt="CleanShot 2023-01-25 at 14 01 35@2x" src="https://user-images.githubusercontent.com/12602440/214658442-d946ae4e-0809-4cba-9b97-414c9526fbcc.png">
<img width="605" alt="CleanShot 2023-01-25 at 14 01 21@2x" src="https://user-images.githubusercontent.com/12602440/214658445-3bc5aee0-8daa-41a0-88ca-d1aa3629c5bf.png">

**Comment by hughess** [comment]
Closing this for now - US map implemented in #595
